analyzer/emerald/emerald.go

```diff
@@ -273,12 +273,11 @@ func (m *Main) queueTransactionInserts(batch *storage.QueryBatch, data *storage.
 		return err
 	}
 	sigContext := signature.DeriveChainContext(rtid, "b11b369e0da5bb230b220127f5e7b242d385ef8c6f54906243f30af63c815535")
-	chainAlias := "mainnet_emerald"
 
 	blockData, err := extractRound(sigContext, data.BlockHeader, data.TransactionsWithResults)
 	if err != nil {
 		return fmt.Errorf("extract round %d: %w", data.Round, err)
 	}
-	emitRoundBatch(batch, chainAlias, int64(data.Round), blockData)
+	emitRoundBatch(batch, int64(data.Round), blockData)
```

Comment:
let's change the in-memory representation to uint64 when we unite the go code

Reply:
postgres doesn't have uint64 so at _some_ point we'll have to do this careful cast and feed the int64 to postgres. (Unless we use `NUMERIC` instead of `BIGINT` in postgres, which we said we wouldn't.)

So I'd recommend implementing the cast here, just so we don't forget, and then we can move it to whatever layer is most appropriate. Your call though.

Reply:
I'm interested in changing the structs imported from nexus to use uint64. then we can have a check right when we're about to save it in the db

Reply:
addendum: in #158 though, as that will be more on the go side

Reply:
Update: I had the same issue in #164 and implemented the check we discussed, but then walked it back because it makes for a crappier debug experience and because I tested that postgres does perform the check. Check the PR description for more info.

So maybe we can handle it the same way here. Your call.

Reply:
thanks for the investigation